### PR TITLE
Add wrapper for LAPACK _trexc methods (reordering of the Schur form)

### PR DESCRIPTION
**Pull request by mfasi**
I need this wrapper to implement some of the points of #5840 (I'm sure I need it for `funm` and `unwm`). I'd be glad to add a few tests, but I have no idea how to test such an operation. Suggestions are welcome.

### CODE REVIEW

File: base/linalg/lapack.jl [hunk 1]
```diff
@@ -3970,10 +3996,36 @@ for (trsen, tgsen, elty) in
...
+            ldq = max(1, stride(Q, 2))
+            work = Array($elty, 1)
+            info = Array(BlasInt, 1)
+
+            ccall(($(blasfunc(trexc)), liblapack), Void,
+                  (Ptr{UInt8},  Ptr{BlasInt},
+                   Ptr{$elty}, Ptr{BlasInt}, Ptr{$elty}, Ptr{BlasInt},
+                   Ptr{BlasInt}, Ptr{BlasInt},
+                   Ptr{$elty}, Ptr{BlasInt}),
+                  &'V', &n,
```
Comment (andreasnoack) [inline base/linalg/lapack.jl]: Usually, we don't restrict the options supported by LAPACK so I'd prefer a version with a `compq` argument. I'd be fine with having a second method without the `compq` argument that calls the first with `compq=='V'`.

File: base/linalg/lapack.jl [hunk 2]
```diff
@@ -3970,10 +3996,36 @@ for (trsen, tgsen, elty) in
...
+#       CHARACTER          COMPQ
+#       INTEGER            IFST, ILST, INFO, LDQ, LDT, N
+# *     ..
+# *     .. Array Arguments ..
+#       DOUBLE PRECISION   Q( LDQ, * ), T( LDT, * ), WORK( * )
+            chkstride1(T, Q)
+            n = chksquare(T)
+            ldt = max(1, stride(T, 2))
+            ldq = max(1, stride(Q, 2))
+            work = Array($elty, 1)
```
Comment (andreasnoack) [inline base/linalg/lapack.jl]: The complex versions don't seem to have a `work` argument. See http://www.netlib.org/lapack/explore-html/dd/ddd/ztrexc_8f.html#a470b89b631cac16cac631539465ae014

File: base/linalg/lapack.jl [hunk 3]
```diff
@@ -3850,10 +3850,36 @@ for (gees, gges, elty, relty) in
...
```
Comment (andreasnoack) [inline base/linalg/lapack.jl]: I think it should have length `n`. See http://www.netlib.org/lapack/explore-html/dc/d51/dtrexc_8f.html#a321d3f3a5e2768676a4d8ec2aa2ff2fe

**Comment by andreasnoack** [comment]
Maybe try one or two reorderings and check that the eigenvalues are unaffected. The main point is to check that the wrapper is working. The two errors I've mentioned would probably have been found with such tests.

**Comment by andreasnoack** [comment]
I think it is fine to add these wrappers, but we should probably not add new functionality to 0.4 so please be patient.

File: base/linalg/lapack.jl (hunk 1 above)
Comment (mfasi) [inline base/linalg/lapack.jl]: I didn't know that, I simply used the `trsen` wrapper as a template. I can fix that one as well. Thank you for having a look at the commit, I'll correct what you suggested and add a few tests shortly.

Comment (andreasnoack) [inline base/linalg/lapack.jl]: Oh. I see. It would be great if you could change that in `trsen` as well and preferably in a separate commit.

**Comment by mfasi** [comment]
@andreasnoack I added the tests you suggested, but some of the checks fail for some reason I cannot understand. For instance, I get a `MethodError` for Linux x86_64, but not for Linux i686. Moreover, the `lapack.jl` test file  works on my machine (Linux x86_64). Is there something I am grossly overlooking?
[I attach the stack trace for reference.]

```
The command "/tmp/julia/bin/julia -e 'versioninfo()'" exited with 0.

127.62s$ export JULIA_CPU_CORES=4 && cd /tmp/julia/share/julia/test && /tmp/julia/bin/julia --check-bounds=yes runtests.jl all && /tmp/julia/bin/julia --check-bounds=yes runtests.jl pkg

	From worker 5:	     * linalg/matmul         in  10.61 seconds

	From worker 5:	     * linalg/schur          in  15.55 seconds

	From worker 5:	     * linalg/special        in   4.40 seconds

	From worker 5:	     * linalg/eigen          in  10.10 seconds

	From worker 5:	     * linalg/bunchkaufman   in   5.24 seconds

	From worker 5:	     * linalg/svd            in   8.56 seconds

	From worker 3:	     * linalg/qr             in  57.18 seconds

	From worker 4:	     * linalg/dense          in  58.58 seconds

	From worker 3:	     * linalg/tridiag        in  14.93 seconds

	From worker 4:	     * linalg/bidiag         in  14.80 seconds

	From worker 2:	     * linalg/triangular     in 120.64 seconds

ERROR: LoadError: On worker 5:

LoadError("/tmp/julia/share/julia/test/linalg/lapack.jl",431,MethodError(trexc!,(1,2,

Float32[11.416281 -1.1872973 1.902823 -2.0817497

        0.0 6.4523597 -0.34416994 2.085003

        0.0 0.0 3.5656812 1.325061

        0.0 0.0 -0.1481058 3.5656812],

Float32[-0.5339776 -0.78375375 0.20146918 -0.24496534

        -0.3179281 0.021552414 -0.9365313 -0.14617346

        -0.75296825 0.5962653 0.27554345 -0.039776057

        -0.21639903 -0.17243204 -0.07997203 0.9576237])))

 in anonymous at no file:435

 in include_string at loading.jl:233

 in include_from_node1 at ./loading.jl:274

 in runtests at /tmp/julia/share/julia/test/testdefs.jl:179

 in anonymous at multi.jl:889

 in run_work_thunk at multi.jl:642

 in anonymous at task.jl:889

while loading /tmp/julia/share/julia/test/runtests.jl, in expression starting on line 13

	From worker 5:	     * linalg/lapack        

The command "export JULIA_CPU_CORES=4 && cd /tmp/julia/share/julia/test && /tmp/julia/bin/julia --check-bounds=yes runtests.jl all && /tmp/julia/bin/julia --check-bounds=yes runtests.jl pkg" exited with 1.
```

**Comment by mfasi** [comment]
Just to record that I solved the problem above. I'll just relax and wait for the end of the feature freeze.

**Comment by kshyatt** [comment]
@mfasi the feature freeze is done since we've branched and this looks possible to merge. Is it ready to go?

**Comment by mfasi** [comment]
@kshyatt I think so, feel free to merge.

**Comment by mfasi** [comment]
@jakebolewski Great, thanks.

**Pull request merged**